Common/include/containers/container_decorators.hpp

```diff
@@ -30,6 +30,43 @@
 
 #include "C2DContainer.hpp"
 
+/*!
+ * \brief Class to represent a matrix (without owning the data, this just wraps a pointer).
+ */
+template<class T>
+class CMatrixView {
+public:
+  using Scalar = typename std::remove_const<T>::type;
+  using Index = unsigned long;
+
+private:
+  T* m_ptr;
+  Index m_cols;
+
+public:
+  CMatrixView(T* ptr = nullptr, Index cols = 0) : m_ptr(ptr), m_cols(cols) {}
+
+  template<class U> friend class CMatrixView;
+  template<class U>
+  CMatrixView(const CMatrixView<U>& other) : m_ptr(other.m_ptr), m_cols(other.m_cols) {}
+
+  explicit CMatrixView(su2matrix<Scalar>& mat) : m_ptr(mat.data()), m_cols(mat.cols()) {}
+
+  template<class U = T, su2enable_if<std::is_const<U>::value> = 0>
+  explicit CMatrixView(const su2matrix<Scalar>& mat) : m_ptr(mat.data()), m_cols(mat.cols()) {}
+
+  const Scalar* operator[] (Index i) const noexcept { return &m_ptr[i*m_cols]; }
+  const Scalar& operator() (Index i, Index j) const noexcept { return m_ptr[i*m_cols + j]; }
+
+  template<class U = T, su2enable_if<!std::is_const<U>::value> = 0>
+  Scalar* operator[] (Index i) noexcept { return &m_ptr[i*m_cols]; }
+
+  template<class U = T, su2enable_if<!std::is_const<U>::value> = 0>
+  Scalar& operator() (Index i, Index j) noexcept { return m_ptr[i*m_cols + j]; }
```

Comment:
Hmmm no..
If the matrix view object is used in a const context then it returns const refs to su2double. Otherwise it will "try" to return non const ref.
However, if the view was returned by a const container, it cannot do that. Hence the use of sfinae to disable the non const version.